### PR TITLE
#9683: add Details Panel for MS dashboard

### PR DESCRIPTION
**Pull request by mahmoudadel54**
## Description
It is a new feature added to MS for dashboard. I added Details panel widget to dashboard. 
- It reflects on Resource grids, dashboard views [View, Edit modes].
- The tool have the same options (eg. show as modal, show at startup etc.)
- The tool is defined in the same way of the corresponding one for maps.
- Edit the layout to put add widget & show/hide connection buttons to the sidebar menu

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue
#9683 

**What is the current behavior?**
#9683 

**What is the new behavior?**
Details panel widget is added for dashboard. User can add/edit dashboard details and set some details for the dashboard and set some options like opening this details on startup, show details on modal/right panel via resource grids and from dashboard view page in view/edit modes.

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Review by MV88 — CHANGES_REQUESTED** [review]
- [ ] there is clearly something wrong here when you create a new dashboard

image

while it works fine when you load a dashboard
image


- [ ] the style of plus button should be blank bg (similar to others) from the start and maybe should be the second one, after login and  when pressed it should become like it is now. because primary background means something active
- [ ] also we should make become a only open button or become unclickable (disabled but visible)

- [ ] i cannot drag and drop or resize widgets on the right if I load this dashboard: http://localhost:8081/?debug=true#/dashboard/21694
image
it works until a certain point notice the cursor, maybe that is the space of details panel which covers click on the area
image
anyway here the placeable area  (red background area) looks properly calculated
image

File: web/client/plugins/SidebarMenu.jsx [hunk 1]
```diff
@@ -229,7 +231,7 @@ class SidebarMenu extends React.Component {
 
     render() {
         return this.state.hidden ? false : (
-            <div id="mapstore-sidebar-menu-container" className="shadow-soft" style={this.getStyle(this.props.style)}>
+            <div id="mapstore-sidebar-menu-container" className={`shadow-soft ${this.props.did ? "fullHightSideBar" : ""}`} style={this.getStyle(this.props.style)}>
```
[the remark this answers is not in the document]
Comment (mahmoudadel54) [reply web/client/plugins/SidebarMenu.jsx:234]: did is dashboard id, but I removed it to make sidebar generic as you said

**Review by MV88 — CHANGES_REQUESTED** [review]
looks better i have only noticed a few minor things:

- [ ] the language selector goes behind body panel, but with a z-index it gets fixed, please test
image
- [ ] the Save button has no tooltip
- [ ] loading property is no longer needed in the DashboardEditor plugin

**Comment by mahmoudadel54** [comment]
> looks better i have only noticed a few minor things:
> 
> * [ ]  the language selector goes behind body panel, but with a z-index it gets fixed, please test
>   image
> * [ ]  the Save button has no tooltip
> * [ ]  loading property is no longer needed in the DashboardEditor plugin

For z-index point, I see that z-index of #mapstore-sidebar-menu-container = 1030 
I have tried to put a z-index to #mapstore-navbar-container with 1031 and it works well so I can specify this style in dashblaod.less to be just for dashboard page only instead of generalize it. 
What do you think ? @MV88

**Comment by MV88** [comment]
> #mapstore-navbar-container

hi if the fix you found is increasing  #mapstore-navbar-container z-index i would not put it inside dashboard.less. but instead in navbar.less and also maybe you can remove the z-index definition of that inside dashboard.less

**Comment by MV88** [comment]
@ElenaGallo please test in DEV

**Comment by ElenaGallo** [comment]
@mahmoudadel54 @tdipisa There are no Export, Import and Delete buttons in the new sidebar.

**QA**
qa

**DEV**
dev

**Comment by tdipisa** [comment]
@mahmoudadel54 I'm sorry but this is of course not correct 
image

this should be fixed to be resource related

**Comment by MV88** [comment]
this has to be backported in c040-2023.02.xx

**Comment by mahmoudadel54** [comment]
> this has to be backported in c040-2023.02.xx

Backport is done --- > https://github.com/geosolutions-it/MapStore2/pull/9750

**Pull request merged**